patches/minecraft/net/minecraft/util/text/ITextComponent.java.patch

```diff
@@ -8,3 +8,19 @@
      String func_150254_d();
  
      List<ITextComponent> func_150253_a();
+@@ -83,6 +82,7 @@
+                     JsonObject jsonobject = p_deserialize_1_.getAsJsonObject();
+                     ITextComponent itextcomponent;
+ 
++                    if ((itextcomponent = net.minecraftforge.common.ForgeHooks.onTextComponentDeserialize(jsonobject)) != null) {/*NOOP*/} else 
```

Comment:
That may work, I'll do that if it's wanted, but I'd feel dirty inside. One of my top rules is to *never* do
```java
if (a)
    if(b) {
        stuff
    }
```
and that's exactly what I'd be doing if I removed that.